### DISCUSSION TITLE
when I bundle esbuild ,but don't allow to do that.

### DISCUSSION THREAD

**Discussion by dddssw** [webpack/webpack#18743] [Q&A]
![image](https://github.com/user-attachments/assets/412e705e-b0e3-406d-93d8-70151396b992)
Is there any configuration for this?

**Comment by alexander-akait** [comment]
Sorry, I don't understand

**Reply by dddssw** [reply]
Sorry for not expressing myself clearly. I want to use the esbuild API to convert ts code into js code. Then use webpack to package esbuild. I don't know if this is feasible.

**Reply by alexander-akait** [reply]
You can use  - https://github.com/privatenumber/esbuild-loader

**Reply by dddssw** [reply]
Not translating TypeScript code during the build process, but at runtime, esbuild loader seems to only work during builds. I know typescript provides an API for converting TS code, but I hope it can be faster. 


```
const ts = require('typescript');


const tsCode = `
const x: number = 1;
console.log(x);
`;


const result = ts.transpileModule(tsCode, {
  compilerOptions: {
    module: ts.ModuleKind.CommonJS,
    target: ts.ScriptTarget.ES2015,
  },
});

console.log(result.outputText); 
```
what I need like the one above  
I don't know if you know of any faster methods than this,Thank very much
